### PR TITLE
feat/get pins

### LINKED ISSUE #63: Add handler for `GET /pins?cids=baf1,...,bafN`
we need to handle batch pin status requests where the list of CIDs to check are passed as a comma separated string of cids under the key `cids` in the URLSearchParams

it's used by the cron-pins scheduled task that runs regularly to poll cluster to see if the pins that were previously Queued are have changed state yet via the cluster client lib `cluster.statusAll`

- cron-pins job where it is called via `cluster.statusAll` - https://github.com/web3-storage/web3.storage/blob/272c83b860c5d15a48c05d3cedb2063b26ca8abe/packages/cron/src/jobs/pins.js#L174
- the implementation of `cluster.statusAll` - https://github.com/nftstorage/ipfs-cluster/blob/02a4c1d0f4cbe86557abd738be83bb2de25ccec0/src/index.js#L189

### PR DESCRIPTION
**Pull request by codeflyer**
Add handler for GET /pins?cids=baf1,...,bafN

closes #63

### CODE REVIEW

File: api/basic/get-pins.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  const chunks: string[][] = []
+
+  for (let i = 0; i < cids.length; i += chunkSize) {
+    const chunk = cids.slice(i, i + chunkSize)
+    chunks.push(chunk)
+  }
+
+  const response: Pin[] = []
+  for (const chunk of chunks) {
+    const data = await client.send(new BatchGetCommand({
```
Comment (simone-sanfratello) [inline api/basic/get-pins.ts:91]: Some questions: how big is the chunk size? Can we set a maximum? If the size is less than 1k, we can get all the keys at once
Anyway, we should run a Promise.all/allSettled

File: api/basic/get-pins.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      RequestItems: {
+        [table]: {
+          Keys: chunk.map(cid => ({ cid }))
+        }
+      },
+      ReturnConsumedCapacity: ReturnConsumedCapacity.TOTAL
+    }))
+
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    data?.Responses?.[table].forEach((item, index) => item ? response.push(item as unknown as Pin) : { cid: chunk[index] })
```
Comment (simone-sanfratello) [inline api/basic/get-pins.ts:101]: What if we get less records than the requested?

File: api/basic/get-pins.ts (hunk 1 above)
Comment (codeflyer) [reply api/basic/get-pins.ts]: The maximum is set by a parameter `chunkSize` (default 50), I've replaced using the Promise.All

File: api/basic/get-pins.ts (hunk 2 above)
Comment (codeflyer) [reply api/basic/get-pins.ts]: It's not a problem, its managed above in the code here:

```
    return {
      statusCode: 200,
      body: cids.map(
        cid => JSON.stringify(toGetPinResponse(cid, pins[cid], ipfsAddr, ipfsPeerId))).join('\n')
    }
```

The result is cerated starting from the cids in the queryparams, and if a value is missing the `toGetPinResponse` manage properly the result.

**Review by simone-sanfratello — APPROVED** [review]

**Pull request merged**